modules/saveload/register_types.cpp

```diff
@@ -0,0 +1,74 @@
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "register_types.h"
+
+#include "saveload_api.h"
+#include "saveload_spawner.h"
+#include "saveload_synchronizer.h"
+#include "scene_saveload.h"
+#include "scene_saveload_interface.h"
+
+#include "saveload_debugger.h"
+
+#ifdef TOOLS_ENABLED
+#include "editor/saveload_editor_plugin.h"
+#endif
+
+static SaveloadAPI *saveload_api = NULL;
+
+void initialize_saveload_module(ModuleInitializationLevel p_level) {
+	if (p_level == MODULE_INITIALIZATION_LEVEL_SERVERS) {
+		saveload_api = memnew(SceneSaveload);
+		GDREGISTER_ABSTRACT_CLASS(SaveloadAPI);
+		Engine::get_singleton()->add_singleton(Engine::Singleton("SaveloadAPI", SaveloadAPI::get_singleton()));
+	}
+	if (p_level == MODULE_INITIALIZATION_LEVEL_SCENE) {
+		GDREGISTER_CLASS(SceneSaveloadConfig);
+		GDREGISTER_CLASS(SaveloadSpawner);
+		GDREGISTER_CLASS(SaveloadSynchronizer);
+		GDREGISTER_CLASS(SceneSaveload);
```

Comment:
I think this too needs to be abstract, as you should never be able to create one

Reply:
Huh, really? All of these are concrete classes though. SaveloadSpawner and SaveloadSynchronizer are nodes. SceneSaveload is an implementation of SaveloadAPI. SceneSaveloadConfig is just for configuring the saveload.

Reply:
I noticed this error in the linux tests.
```
/home/runner/work/godot/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/SceneSaveload.cs(11,22): error CS0709: 'SceneSaveload': cannot derive from static class 'SaveloadApi' [/home/runner/work/godot/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
```
What even is a static class? 😅

Reply:
The `SceneSaveload` specifically

Reply:
Because `SceneSaveload` implements the abstract `SaveloadAPI`? I'll try it!

Thanks for all the support with the CI ❤️